src/main/java/org/jabref/logic/search/SearchQuery.java

```diff
@@ -133,7 +148,7 @@ public boolean isRegularExpression() {
         // compile the words to a regular expression in the form (w1)|(w2)|(w3)
         StringJoiner joiner = new StringJoiner(")|(", "(", ")");
         for (String word : words) {
-            joiner.add(regularExpression ? word : Pattern.quote(word));
+            joiner.add(regularExpression ? word : (escapeSpecialCharsForJS ? word.replaceAll(JAVASCRIPT_ESCAPED_CHARS_PATTERN, "\\\\$0") : Pattern.quote(word)));
```

Comment:
Please extract this to a regular if-else, it's easier to understand on the first look than this chained conditions